app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorView.kt

```diff
@@ -23,11 +27,49 @@ import org.mozilla.fenix.settings.address.interactor.AddressEditorInteractor
 class AddressEditorView(
     private val binding: FragmentAddressEditorBinding,
     private val interactor: AddressEditorInteractor,
+    private val region: RegionState?,
     private val address: Address? = null
 ) {
 
     /**
-     * Binds the view.
+     * Value type representing properties determined by the country used in an [Address].
+     * This data is meant to mirror the data currently represented on desktop here:
+     * https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/addressmetadata/addressReferences.js
+     *
+     * This can be expanded to included things like a list of applicable states/provinces per country
+     * or the names that should be used for each form field.
+     *
+     * Note: Most properties here need to be kept in sync with the data in the above desktop
+     * address reference file in order to prevent duplications when sync is enabled. There are
+     * ongoing conversations about how best to share that data cross-platform, if at all.
+     * Some more detail: https://bugzilla.mozilla.org/show_bug.cgi?id=1769809
+     *
+     * Exceptions: [displayName] is a local property and stop-gap to a more robust solution.
+     *
+     * @property key The country code used to lookup the address data. Should match desktop entries.
+     * @property name The full name of the region. Should match desktop entries.
+     * @property displayName The name to display when selected.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal data class ValidAddressCountry(
+        val key: String,
+        val name: String,
+        val displayName: String,
+    )
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal val defaultCountry = ValidAddressCountry(
```

Comment:
This is just a feeling from reading everything, but I almost feel like it could be suffice to use a constant country code `const val DEFAULT_COUNTRY = "US"` to avoid having an entire `ValidAddressCountry` appear out of place in the map.

Reply:
This was mostly to avoid map lookups, which are nullable. For example, this change would mean making substantial changes to tests. Instead of 

```
assertEquals(addressEditorView.defaultCountry.displayName, binding.countryDropDown.selectedItem.toString())
```

it would require something like 

```
assertEquals(addressEditorView.countries[DEFAULT_COUNTRY]!!.displayName, binding.countryDropDown.selectedItem.toString())
```

I'm fine with using the unsafe accessor in tests, but it may mean additional null guards in regular code as well.

Reply:
Makes sense. I am also fine with using the unsafe accessor in tests. I can imagine that when we do have an API that provides a similar map or object that we wouldn't necessarily be able to avoid the map lookup or null guards. 

I am okay with it as is. I will leave the final choice to you.

Reply:
I agree that it makes the initial definition much cleaner, so I went with your suggestion